src/engine/controls/cuecontrol.cpp

```diff
@@ -1764,12 +1765,12 @@ void CueControl::hotcueFocusColorPrev(double value) {
         return;
     }
 
-    int hotcueNumber = static_cast<int>(m_pHotcueFocus->get());
-    if (hotcueNumber < 0 || hotcueNumber >= m_hotcueControls.size()) {
+    int hotcueIndex = static_cast<int>(m_pHotcueFocus->get()) - 1;
```

Comment:
Please define a function to map hot cue control numbers to the corresponding index using the kNoHotCue and kFirstHotCue constants appropriately.